base/tuple.jl

```diff
@@ -27,7 +27,8 @@ lastindex(@nospecialize t::Tuple) = length(t)
 size(@nospecialize(t::Tuple), d::Integer) = (d == 1) ? length(t) : throw(ArgumentError("invalid tuple dimension $d"))
 axes(@nospecialize t::Tuple) = (OneTo(length(t)),)
 @eval getindex(@nospecialize(t::Tuple), i::Int) = getfield(t, i, $(Expr(:boundscheck)))
-@eval getindex(@nospecialize(t::Tuple), i::Real) = getfield(t, convert(Int, i), $(Expr(:boundscheck)))
+@eval getindex(@nospecialize(t::Tuple), i::Integer) = getfield(t, convert(Int, i), $(Expr(:boundscheck)))
+@deprecate @eval getindex(@nospecialize(t::Tuple), i::Real) = getfield(t, convert(Int, i), $(Expr(:boundscheck)))
```

Comment:
Should move to deprecated.jl, and I think the eval will have to go outside since deprecate expects function syntax.

Reply:
```suggestion
@deprecate getindex(@nospecialize(t::Tuple), i::Real) = t[convert(Int, i)]
```

(this will make the deprecation message better too)

Reply:
But this also needs to go in the deprecated.jl file, not here